src/Containers/container.jl

```diff
@@ -163,3 +163,16 @@ function _sparseaxisarray(dict::Dict{Any,Any}, f, indices)
     d = _container_dict(_default_eltype(indices), f, _eltype_or_any(indices))
     return SparseAxisArray(d)
 end
+
+function container(f::Function, indices, D::Type{<:AbstractDict})
+    # Don't use length-1 tuples if there is only one index!
+    key(i) = length(i) == 1 ? i[1] : i
```

Comment:
Is this type stable ? Why is this defined inside `container` ? We could have a function defined outside with two method and detect that it's a tuple with one element in the signature to ensure type stability.